util/stop/stopper_test.go

```diff
@@ -243,9 +242,14 @@ func TestStopperClosers(t *testing.T) {
 func TestStopperNumTasks(t *testing.T) {
 	s := stop.NewStopper()
 
-	var tasks []stop.PendingTask
+	var tasks []bool
+	var wg sync.WaitGroup
 	for i := 0; i < 3; i++ {
-		tasks = append(tasks, s.StartTask())
+		wg.Add(1)
+		tasks = append(tasks, s.RunGoTask(func() {
+			time.Sleep(5 * time.Millisecond)
```

Comment:
Instead of sleeping, we should have a slice of channels that we can close down where the old code called task.Done.